src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -367,43 +372,37 @@ internal FunctionActivityTracker(IOptionsMonitor<AppServiceOptions> appServiceOp
                     _metricsPublisher = metricsPublisher;
                 }
 
-                StartActivityTimer();
+                _activityFlushCounter = _functionActivityFlushInterval;
+                _activityTimer = new Timer(TimerFlush, null, _activityTimerIntervalMS, _activityTimerIntervalMS);
```

Comment:
Any idea why this wasn't a Timer to begin with?

Reply:
Not sure